libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/targets/js/npm/NpmDependencyExtension.kt

```diff
@@ -19,82 +19,102 @@ interface NpmDependencyExtension {
+        return when (arg) {
+            is String -> withName(
+                name = arg,
+                args = *args
+            )
+            is File -> invoke(arg)
+            else -> throw npmDeclarationException(args)
         }
+    }
 
-        private fun withName(name: String, vararg args: Any?): NpmDependency {
-            val arg = if (args.size > 1) args[1] else null
-            return when (arg) {
-                null -> invoke(
-                    name = name
-                )
-                is String -> invoke(
-                    name = name,
-                    version = arg
-                )
-                is File -> invoke(
-                    name = name,
-                    directory = arg
-                )
-                else -> throw npmDeclarationException(args)
-            }
+    private fun withName(name: String, vararg args: Any?): NpmDependency {
+        val arg = if (args.size > 1) args[1] else null
+        return when (arg) {
+            null -> invoke(
+                name = name
+            )
+            is String -> invoke(
+                name = name,
+                version = arg
+            )
+            is File -> invoke(
+                name = name,
+                directory = arg
+            )
+            else -> throw npmDeclarationException(args)
         }
+    }
 
-        private fun npmDeclarationException(args: Array<out Any?>): IllegalArgumentException {
-            return IllegalArgumentException(
-                """
+    private fun npmDeclarationException(args: Array<out Any?>): IllegalArgumentException {
+        return IllegalArgumentException(
+            """
                             Unable to add NPM dependency by $args
                             - npm('name') -> name:*
                             - npm('name', 'version') -> name:version
                             - npm('name', File) -> name:File
                             """.trimIndent()
-            )
-        }
+        )
     }
+}
 
-    dependencies
-        .extensions
-        .add(
-            TypeOf.typeOf<NpmDependencyExtension>(NpmDependencyExtension::class.java),
-            "npm",
-            npmDependencyExtension
-        )
-}
\ No newline at end of file
+private const val NAME_DEPENDENCIES = "npm"
+private const val NAME_DEV_DEPENDENCIES = "devNpm"
+private const val NAME_PEER_DEPENDENCIES = "peerNpm"
+
+fun Project.addNpmDependencyExtension() {
+    val dependencies = this.dependencies as ExtensionAware
+
+    listOf(NpmDependency.Scope.NORMAL, NpmDependency.Scope.DEV, NpmDependency.Scope.PEER).forEach { scope ->
+        val name = when (scope) {
+            NpmDependency.Scope.DEV -> NAME_DEV_DEPENDENCIES
+            NpmDependency.Scope.PEER -> NAME_PEER_DEPENDENCIES
+            else -> NAME_DEPENDENCIES
+        }
+
+        val dependencyExtension: NpmDependencyExtension = object : AbstractNpmDependencyExtension(dependencies, this, scope) {}
```

Comment:
Why is it `abstract` if it doesn't implement any abstract members?
It can be usual class `DefaultNpmDependencyExtension`